src/main/java/com/github/wovnio/wovnjava/Settings.java

```diff
@@ -96,6 +100,17 @@
         this.outboundProxyPort = reader.getIntParameter("outboundProxyPort");
 
         this.encoding = stringOrDefault(reader.getStringParameter("encoding"), "");
+
+        this.fixedHost = stringOrDefault(reader.getStringParameter("fixedHost"), "");
+        this.fixedScheme = stringOrDefault(reader.getStringParameter("fixedScheme"), "");
+        this.fixedPort = positiveIntOrDefault(reader.getIntParameter("fixedPort"), -1);
+        this.verifyFixedURLConfigs(this.fixedHost, this.fixedScheme, this.fixedPort);
+    }
+
+    private void verifyFixedURLConfigs(String fixedHost, String fixedScheme, int fixedPort) throws ConfigurationError {
+        if ( (fixedHost.isEmpty() != fixedScheme.isEmpty()) || (fixedScheme.isEmpty() != (fixedPort == -1))) {
```

Comment:
This looks very confusing and I have to think about it a lot. Why are we comparing these against each other? I think there should be a better way to write it.

Reply:
For example
1. Check if all parameters are empty or -1, return true if so
2. Otherwise assert `notEmpty && notEmpt && port != -1`

Reply:
We want to test if three things a, b, c are the same. The most efficient form would be `a == b && b == c`. And to negate it it becomes `a != b || b != c`. 

I think I can use the `a == b && b == c` form and do an early return if it's the case. That should be simpler. What do you think?

Reply:
we should optimize for readability here, we don't need to worry about a few extra boolean comparisons. Extract variables if you want, the compiler doesn't care if we split it up over multiple lines.

Reply:
the way you've written it is more verbose than it could be, which isn't helping. If you want to assert three variables are equal you can do `a == b == c` and not need conditionals

Reply:
`a == b == c` wouldn't work would it, because it evaluates from left to right. `false == false == true` would be `true`.

Reply:
ok yeah, but this is exactly why we shouldn't make code too clever for the sake of efficiency and have to think about what it means 😛

Reply:
OK I like it now. Just having an explicit variable name is helpful.